go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -66,6 +67,41 @@ func (pbr *PolicyBasedRoutesManager) AddSameNodeIPPolicy(nodeName, mgmtPortIP st
 	return nil
 }
 
+// AddHostCIDRPolicy adds the following policy in local-gateway-mode for L2 topology
+// 2000  ip4.dst==172.18.0.0/16   reroute    10.128.0.2
+// Since rtoe of GR is directly connected to the hostCIDR range in LGW even with the
+// reroute to mp0 src-ip route the dst-ip based default OVN route takes precedence and
```

Comment:
so its the directly connected route in the GR connected to 172.18.0.0 that is overriding your src-ip route? Should you match on src traffic being the pod subnet? I'm wondering if ovn-controller injecting control plane traffic back into OVS if it would reroute it towards mp0. There is an issue from the past similar to this. I'll try to remember it.

Reply:
can you print the routes from ovn_cluster_router in this scenario? Would also be good to add that and more explanation to the description.

Reply:
yes let me give a snapshot of this on the GR (since its L2)

Reply:
> so its the directly connected route in the GR connected to 172.18.0.0 that is overriding your src-ip route? 

yeah exactly!

> Should you match on src traffic being the pod subnet? 

yea I thought about this as well, do you think just have a single LRP rerouting all of the traffic with podIP==srcIP is better than having a LRSR? I was trying to keep it as close as possible to L3 and wasn't sure if doing the LRSR will break other things or not

Reply:
```
sh-5.2# ovn-nbctl lr-route-list GR_l2.network_ovn-worker2
IPv4 Routes
Route Table <main>:
          10.100.200.0/24              10.100.200.2 src-ip
           169.254.0.0/17               169.254.0.4 dst-ip rtoe-GR_l2.network_ovn-worker2
                0.0.0.0/0                172.18.0.1 dst-ip rtoe-GR_l2.network_ovn-worker2
sh-5.2# ovn-nbctl lr-policy-list GR_l2.network_ovn-worker2                                                                                                                   
Routing Policies
      1500                           ip4.dst == 172.18.0.0/16         reroute              10.100.200.2
      1004 inport == "rtos-l2.network_ovn_layer2_switch" && ip4.dst == 172.18.0.4 /* l2.network_ovn_layer2_switch */         reroute              10.100.200.2
```

Reply:
made the explanation better lmk how it is now.

Reply:
as we discussed, ovn-controller generated traffic may be affected so we need to match the src ip subnet as pod subnet. Also, we need to lower the priority to <100 so that we dont override egress IP LRPs. Some of those will need to be updated to reroute to mp0 once egress IP is implemented for lgw mode @martinkennelly fyi

Reply:
this is done in the recent push